README.md
# Twitter & OECD
Here you can find working crawlers and prototypes being tested for the CUTLER platform. Crawlers are tested with mongodb://localhost. Users should provide own Access tokens (CONSUMER_KEY, CONSUMER_SECRET,ACCESS_TOKEN, ACCESS_TOKEN_SECRET) that can be obtained via https://apps.twitter.com/

**List of Crawlers**

**OECD:** Reuses the OECD library to crawl the series requested from the Economic WP.

**Twitter/TwGeoBox.py:** Collects all streaming tweets from a given location. The location is defined as GEOBOX_CHICAGO = [-87.9671519386,41.62241735,-87.3972361672,42.036640212]. Returns JSON object. 

**Twitter/Timelines.py:** Retrieves recent tweets (and other information from the original JSON object) from a static list of known Twitter users. Returns a 2D array with predetermined fields from the JSON object. 

**Twitter/Twitter_keywords.py:** Retrieves all streaming tweets that contain one or more keywords from a static list. Returns JSON object.



news.md
Documentation on how to use OECD data for the CUTLER project



The OECD.stat database has a breakdown of 23 themes totalling 1245 series. 

#List all codes and descriptions of available OECD series.
dataset_list <- get_datasets()


Each theme has additional branches. We are interested in the Regions and Cities (Theme 20) that contains 4 branches (A,B,C,D). We are particularly interested in the first branch: Regional Statistics. Within this branch 
we are collecting two series: i) Regional Economy(REGION_ECONOM) and ii) Regional Labour (REGION_LABOUR).

Description of the available values for dimension VAR in datasets REGION_ECONOM and REGION_LABOUR are here: http://sdmx.herokuapp.com/oecd/codelist/VAR?Dataset=REGION_LABOUR and http://sdmx.herokuapp.com/oecd/codelist/VAR?Dataset=REGION_ECONOM

Also further vocs can be found here: https://www.econdb.com/source/OECD/organisation-for-economic-co-operation-and-development/


1.	General Statistics
2.	Agriculture and Fisheries
3.	Demography and Population
4.	Development
5.	Economic Projections
6.	Education and Training
7.	Environment
8.	Finance
9.	Globalisation
10.	Health
11.	Industry and Services
12.	Information and Communication Technology
13.	International Trade and Balance of Payments
14.	Labour
15.	National Accounts
16.	Monthly Economic Indicators
17.	Prices and Purchasing Power Parities
18.	Public Sector, Taxation and Market Regulation
19.	Productivity
20.	Regions and Cities
		A. Regional Statistics
			1. Database on Migrants in OECD Regions
			2. Regional Business Demography
			3. Regional Demography
			4. Regional Economy (id = REGION_ECONOM)
				Regional Gross Domestic Product (Large regions TL2)
				Regional Gross Domestic Product (Small regions TL3)
				Regional GVA by industry (ISIC rev 4)
				Regional Employment by industry (ISIC rev 4)
				Regional GVA per worker
				Labour utilisation rate
				Regional income per capita
				Regional income per equivalised household
				Average annual population (used for per capita measures)
				Reference series - deflators and PPP rates
				Economic indicators by Typology (rural/urban)

			5. Regional Labour (id = REGION_LABOUR)
				Labour indicators (Large regions TL2)
				Employment at place of residence
				Part-Time Employment
				Youth and Long Term Unemployment
				Highest and Lowest regional values by TL2
				Labour Force and Participation rate
				Unemployment in regions
			
			6. Regional Social and Environmental indicators
			7. Regional Innovation

		B. Regional Well-Being
		C. Metropolitan areas
		D. Subnational Government Structure and Finance
21.	Science, Technology and Patents
22.	Social Protection and Well-being
23.	Transport





library(OECD)
dataset_list <- get_datasets()

#REGION_LABOUR 
Greece <- "1+2+3_PU+3_IN+3_PR+NOG.GRC+EL30+EL301+EL302+EL303+EL304+EL305+EL306+EL307+EL41+EL411+EL412+EL413+EL42+EL421+EL422+EL43+EL431+EL432+EL433+EL434+EL51+EL511+EL512+EL513+EL514+EL515+EL52+EL521+EL522+EL523+EL524+EL525+EL526+EL527+EL53+EL531+EL532+EL533+EL54+EL541+EL542+EL543+EL61+EL611+EL612+EL613+EL62+EL621+EL622+EL623+EL624+EL63+EL631+EL632+EL633+EL64+EL641+EL642+EL643+EL644+EL645+EL65+EL651+EL652+EL653"
Ireland <- "1+2+3_PU+3_IN+3_PR+NOG.IRL+IE01+IE011+IE012+IE013+IE02+IE021+IE022+IE023+IE024+IE025"
Turkey <-"1+2+3_PU+3_IN+3_PR+NOG.TUR+TR10+TR100+TR21+TR211+TR212+TR213+TR22+TR221+TR222+TR31+TR310+TR32+TR321+TR322+TR323+TR33+TR331+TR332+TR333+TR334+TR41+TR411+TR412+TR413+TR42+TR421+TR422+TR423+TR424+TR425+TR51+TR510+TR52+TR521+TR522+TR61+TR611+TR612+TR613+TR62+TR621+TR622+TR63+TR631+TR632+TR633+TR71+TR711+TR712+TR713+TR714+TR715+TR72+TR721+TR722+TR723+TR81+TR811+TR812+TR813+TR82+TR821+TR822+TR823+TR83+TR831+TR832+TR833+TR834+TR90+TR901+TR902+TR903+TR904+TR905+TR906+TRA1+TRA11+TRA12+TRA13+TRA2+TRA21+TRA22+TRA23+TRA24+TRB1+TRB11+TRB12+TRB13+TRB14+TRB2+TRB21+TRB22+TRB23+TRB24+TRC1+TRC11+TRC12+TRC13+TRC2+TRC21+TRC22+TRC3+TRC31+TRC32+TRC33+TRC34"
Belgium <- "1+2+3_PU+3_IN+3_PR+NOG.BEL+BE1+BE100+BE2+BE211+BE212+BE213+BE221+BE222+BE223+BE231+BE232+BE233+BE234+BE235+BE236+BE241+BE242+BE251+BE252+BE253+BE254+BE255+BE256+BE257+BE258+BE3+BE310+BE321+BE322+BE323+BE324+BE325+BE326+BE327+BE331+BE332+BE334+BE335+BE336+BE341+BE342+BE343+BE344+BE345+BE351+BE352+BE353"
df_REGION_LABOUR_GR <- get_dataset("REGION_LABOUR", filter = Greece , pre_formatted = TRUE)
df_REGION_LABOUR_IR <- get_dataset("REGION_LABOUR", filter = Ireland , pre_formatted = TRUE)
df_REGION_LABOUR_TU <- get_dataset("REGION_LABOUR", filter = Turkey , pre_formatted = TRUE)
df_REGION_LABOUR_BE <- get_dataset("REGION_LABOUR", filter = Belgium , pre_formatted = TRUE)


Greece <- "1+2+3_PU+3_IN+3_PR+NOG.GRC+EL30+EL301+EL302+EL303+EL304+EL305+EL306+EL307+EL41+EL411+EL412+EL413+EL42+EL421+EL422+EL43+EL431+EL432+EL433+EL434+EL51+EL511+EL512+EL513+EL514+EL515+EL52+EL521+EL522+EL523+EL524+EL525+EL526+EL527+EL53+EL531+EL532+EL533+EL54+EL541+EL542+EL543+EL61+EL611+EL612+EL613+EL62+EL621+EL622+EL623+EL624+EL63+EL631+EL632+EL633+EL64+EL641+EL642+EL643+EL644+EL645+EL65+EL651+EL652+EL653"
Ireland <- "1+2+3_PU+3_IN+3_PR+NOG.IRL+IE01+IE011+IE012+IE013+IE02+IE021+IE022+IE023+IE024+IE025"
Turkey <-"1+2+3_PU+3_IN+3_PR+NOG.TUR+TR10+TR100+TR21+TR211+TR212+TR213+TR22+TR221+TR222+TR31+TR310+TR32+TR321+TR322+TR323+TR33+TR331+TR332+TR333+TR334+TR41+TR411+TR412+TR413+TR42+TR421+TR422+TR423+TR424+TR425+TR51+TR510+TR52+TR521+TR522+TR61+TR611+TR612+TR613+TR62+TR621+TR622+TR63+TR631+TR632+TR633+TR71+TR711+TR712+TR713+TR714+TR715+TR72+TR721+TR722+TR723+TR81+TR811+TR812+TR813+TR82+TR821+TR822+TR823+TR83+TR831+TR832+TR833+TR834+TR90+TR901+TR902+TR903+TR904+TR905+TR906+TRA1+TRA11+TRA12+TRA13+TRA2+TRA21+TRA22+TRA23+TRA24+TRB1+TRB11+TRB12+TRB13+TRB14+TRB2+TRB21+TRB22+TRB23+TRB24+TRC1+TRC11+TRC12+TRC13+TRC2+TRC21+TRC22+TRC3+TRC31+TRC32+TRC33+TRC34"

OECD.r
library(OECD)

#Define our Locations (L3)
PiThess <- "1+2+3_PU+3_IN+3_PR+NOG.GRC+EL527"
PiCork <- "1+2+3_PU+3_IN+3_PR+NOG.IRL+IE025"
PiAntw <- "1+2+3_PU+3_IN+3_PR+NOG.BEL+BE211"
PiAntal <-"1+2+3_PU+3_IN+3_PR+NOG.TUR+TR611"

#Define the economic Indicators that we need to crawl. 
Eco_OECD_RegionLabour_EmpPlaRes <- "EMP_Y15_MAX+EMP_Y15_64+EMP_Y15_24+EMP_RA_15_MAX+EMP_RA_15_64+EMP_RA_15_MAX_GR_2001+EMP_RA_15_64_GR_2001+EMP_RA_15_MAX_GR_2007+EMP_RA_15_64_GR_2007+EMP_RA_15_64_SEXDIF+EMP_RA_15_MAX_SEXDIF+EMP_PT+EMP_PT_SH+EMP_PT_SH_SEXDIF.T+F+M.ALL"
Eco_OECD_RegionLabour_LfPartRa <- "REG_LFS+POP_Y15_MAX+POP_Y15_64+LF_Y15_MAX+LF_Y15_64+LF_Y15_24+PARTIC_RA_15_MAX+PARTIC_RA_15_64+PARTIC_RA_15_MAX_GR_2001+PARTIC_RA_15_64_GR_2001+PARTIC_RA_15_MAX_GR_2007+PARTIC_RA_15_64_GR_2007+PARTIC_RA_15_MAX_SEXDIF+PARTIC_RA_15_64_SEXDIF.T+F+M.ALL"
Eco_OECD_RegionLabour_UnemReg <- "UNEM+UNEM_RA_15_MAX+UNEM_RA_15_64+UNEM_RA_15_MAX_GR_2007+UNEM_RA_15_64_GR_2007+UNEM_RA_15_MAX_SEXDIF+UNEM_RA_15_64_SEXDIF.T+F+M.ALL"
Eco_OECD_RegionEco_RegGdpTL2 <- "SNA_2008.GDP.REG+CURR_PR+USD_PPP+REAL_PR+REAL_PPP+PC+PC_CURR_PR+PC_USD_PPP+PC_REAL_PR+PC_REAL_PPP+GWTH_2007+GWTH_REAL_PR_2007+GWTH_PC_REAL_PR_2007+GWTH_2001+GWTH_REAL_PR_2001+GWTH_PC_REAL_PR_2001.ALL"
Eco_OECD_RegionEco_GDPLT3 <- "SNA_2008.GDP.REG+CURR_PR+USD_PPP+REAL_PR+REAL_PPP+PC+PC_CURR_PR+PC_USD_PPP+PC_REAL_PR+PC_REAL_PPP+GWTH_2007+GWTH_REAL_PR_2007+GWTH_PC_REAL_PR_2007+GWTH_2001+GWTH_REAL_PR_2001+GWTH_PC_REAL_PR_2001.ALL"
Eco_OECD_RegionEco_RegEmIndu <- "SNA_2008.EMP_IND_TOTAL+EMP_IND_10_VA+EMP_IND_10_VC+EMP_IND_10_VB_E+EMP_IND_10_VF+EMP_IND_10_VG_I+EMP_IND_10_VJ+EMP_IND_10_VK+EMP_IND_10_VL+EMP_IND_10_VM_N+EMP_IND_10_VO_Q+EMP_IND_10_VR_U.PER.ALL"
Eco_OECD_RegionEco_RegGVAWorker <- "SNA_2008.GVA_IND_TOTAL+GVA_IND_10_VA+GVA_IND_10_VC+GVA_IND_10_VB_E+GVA_IND_10_VF+GVA_IND_10_VG_I+GVA_IND_10_VJ+GVA_IND_10_VK+GVA_IND_10_VL+GVA_IND_10_VM_N+GVA_IND_10_VO_Q+GVA_IND_10_VR_U.PW_CURR_PR+PW_USD_PPP+PW_REAL_PR+PW_REAL_PPP.ALL"
Eco_OECD_RegionEco_RegIncPC <- "SNA_2008.INCOME_DISP+INCOME_PRIM.PC_CURR_PR+PC_USD_PPP+PC_REAL_PR+PC_REAL_PPP.ALL"

# Define the filters of the request by combining locations and indicators.  
filter_PiThess_Eco_OECD_RegionLabour_EmpPlaRes <-paste(PiThess, Eco_OECD_RegionLabour_EmpPlaRes,sep=".")
filter_PiAntw_Eco_OECD_RegionLabour_EmpPlaRes <-paste(PiAntw, Eco_OECD_RegionLabour_EmpPlaRes,sep=".")
filter_PiCork_Eco_OECD_RegionLabour_EmpPlaRes <-paste(PiCork, Eco_OECD_RegionLabour_EmpPlaRes,sep=".")
filter_PiAntal_Eco_OECD_RegionLabour_EmpPlaRes <-paste(PiAntal, Eco_OECD_RegionLabour_EmpPlaRes,sep=".")
filter_PiThess_Eco_OECD_RegionLabour_LfPartRa <-paste(PiThess, Eco_OECD_RegionLabour_LfPartRa,sep=".")
filter_PiAntw_Eco_OECD_RegionLabour_LfPartRa <-paste(PiAntw, Eco_OECD_RegionLabour_LfPartRa,sep=".")
filter_PiCork_Eco_OECD_RegionLabour_LfPartRa <-paste(PiCork, Eco_OECD_RegionLabour_LfPartRa,sep=".")
filter_PiAntal_Eco_OECD_RegionLabour_LfPartRa <-paste(PiAntal, Eco_OECD_RegionLabour_LfPartRa,sep=".")
filter_PiThess_Eco_OECD_RegionLabour_UnemReg <-paste(PiThess, Eco_OECD_RegionLabour_UnemReg,sep=".")
filter_PiAntw_Eco_OECD_RegionLabour_UnemReg <-paste(PiAntw, Eco_OECD_RegionLabour_UnemReg,sep=".")
filter_PiCork_Eco_OECD_RegionLabour_UnemReg <-paste(PiCork, Eco_OECD_RegionLabour_UnemReg,sep=".")
filter_PiAntal_Eco_OECD_RegionLabour_UnemReg <-paste(PiAntal, Eco_OECD_RegionLabour_UnemReg,sep=".")
filter_PiThess_Eco_OECD_RegionEco_RegGdpTL2 <-paste(PiThess, Eco_OECD_RegionEco_RegGdpTL2,sep=".")
filter_PiAntw_Eco_OECD_RegionEco_RegGdpTL2 <-paste(PiAntw, Eco_OECD_RegionEco_RegGdpTL2,sep=".")
filter_PiCork_Eco_OECD_RegionEco_RegGdpTL2 <-paste(PiCork, Eco_OECD_RegionEco_RegGdpTL2,sep=".")
filter_PiAntal_Eco_OECD_RegionEco_RegGdpTL2 <-paste(PiAntal, Eco_OECD_RegionEco_RegGdpTL2,sep=".")
filter_PiThess_Eco_OECD_RegionEco_GDPLT3 <-paste(PiThess, Eco_OECD_RegionEco_GDPLT3,sep=".")
filter_PiAntw_Eco_OECD_RegionEco_GDPLT3 <-paste(PiAntw, Eco_OECD_RegionEco_GDPLT3,sep=".")
filter_PiCork_Eco_OECD_RegionEco_GDPLT3 <-paste(PiCork, Eco_OECD_RegionEco_GDPLT3,sep=".")
filter_PiAntal_Eco_OECD_RegionEco_GDPLT3 <-paste(PiAntal, Eco_OECD_RegionEco_GDPLT3,sep=".")
filter_PiThess_Eco_OECD_RegionEco_RegEmIndu <-paste(PiThess, Eco_OECD_RegionEco_RegEmIndu,sep=".")
filter_PiAntw_Eco_OECD_RegionEco_RegEmIndu <-paste(PiAntw, Eco_OECD_RegionEco_RegEmIndu,sep=".")
filter_PiCork_Eco_OECD_RegionEco_RegEmIndu <-paste(PiCork, Eco_OECD_RegionEco_RegEmIndu,sep=".")
filter_PiAntal_Eco_OECD_RegionEco_RegEmIndu <-paste(PiAntal, Eco_OECD_RegionEco_RegEmIndu,sep=".")
filter_PiThess_Eco_OECD_RegionEco_RegGVAWorker <-paste(PiThess, Eco_OECD_RegionEco_RegGVAWorker,sep=".")
filter_PiAntw_Eco_OECD_RegionEco_RegGVAWorker <-paste(PiAntw, Eco_OECD_RegionEco_RegGVAWorker,sep=".")
filter_PiCork_Eco_OECD_RegionEco_RegGVAWorker <-paste(PiCork, Eco_OECD_RegionEco_RegGVAWorker,sep=".")
filter_PiAntal_Eco_OECD_RegionEco_RegGVAWorker <-paste(PiAntal, Eco_OECD_RegionEco_RegGVAWorker,sep=".")
filter_PiThess_Eco_OECD_RegionEco_RegIncPC <-paste(PiThess, Eco_OECD_RegionEco_RegIncPC,sep=".")
filter_PiAntw_Eco_OECD_RegionEco_RegIncPC <-paste(PiAntw, Eco_OECD_RegionEco_RegIncPC,sep=".")
filter_PiCork_Eco_OECD_RegionEco_RegIncPC <-paste(PiCork, Eco_OECD_RegionEco_RegIncPC,sep=".")
filter_PiAntal_Eco_OECD_RegionEco_RegIncPC <-paste(PiAntal, Eco_OECD_RegionEco_RegIncPC,sep=".")

#Make the request using the "get_dataset" function and save the data in unique dataframes. 
PiThess_Eco_OECD_RegionLabour_EmpPlaRes <-get_dataset("REGION_LABOUR", filter=filter_PiThess_Eco_OECD_RegionLabour_EmpPlaRes, pre_formatted = TRUE)
PiAntw_Eco_OECD_RegionLabour_EmpPlaRes <-get_dataset("REGION_LABOUR", filter=filter_PiAntw_Eco_OECD_RegionLabour_EmpPlaRes, pre_formatted = TRUE)
PiCork_Eco_OECD_RegionLabour_EmpPlaRes <-get_dataset("REGION_LABOUR", filter=filter_PiCork_Eco_OECD_RegionLabour_EmpPlaRes, pre_formatted = TRUE)
PiAntal_Eco_OECD_RegionLabour_EmpPlaRes <-get_dataset("REGION_LABOUR", filter=filter_PiAntal_Eco_OECD_RegionLabour_EmpPlaRes, pre_formatted = TRUE)
PiThess_Eco_OECD_RegionLabour_LfPartRa <-get_dataset("REGION_LABOUR", filter=filter_PiThess_Eco_OECD_RegionLabour_LfPartRa, pre_formatted = TRUE)
PiAntw_Eco_OECD_RegionLabour_LfPartRa <-get_dataset("REGION_LABOUR", filter=filter_PiAntw_Eco_OECD_RegionLabour_LfPartRa, pre_formatted = TRUE)
PiCork_Eco_OECD_RegionLabour_LfPartRa <-get_dataset("REGION_LABOUR", filter=filter_PiCork_Eco_OECD_RegionLabour_LfPartRa, pre_formatted = TRUE)
PiAntal_Eco_OECD_RegionLabour_LfPartRa <-get_dataset("REGION_LABOUR", filter=filter_PiAntal_Eco_OECD_RegionLabour_LfPartRa, pre_formatted = TRUE)
PiThess_Eco_OECD_RegionLabour_UnemReg <-get_dataset("REGION_LABOUR", filter=filter_PiThess_Eco_OECD_RegionLabour_UnemReg, pre_formatted = TRUE)
PiAntw_Eco_OECD_RegionLabour_UnemReg <-get_dataset("REGION_LABOUR", filter=filter_PiAntw_Eco_OECD_RegionLabour_UnemReg, pre_formatted = TRUE)
PiCork_Eco_OECD_RegionLabour_UnemReg <-get_dataset("REGION_LABOUR", filter=filter_PiCork_Eco_OECD_RegionLabour_UnemReg, pre_formatted = TRUE)
PiAntal_Eco_OECD_RegionLabour_UnemReg <-get_dataset("REGION_LABOUR", filter=filter_PiAntal_Eco_OECD_RegionLabour_UnemReg, pre_formatted = TRUE)
PiThess_Eco_OECD_RegionEco_RegGdpTL2 <-get_dataset("REGION_ECONOM", filter=filter_PiThess_Eco_OECD_RegionEco_RegGdpTL2, pre_formatted = TRUE)
PiAntw_Eco_OECD_RegionEco_RegGdpTL2 <-get_dataset("REGION_ECONOM", filter=filter_PiAntw_Eco_OECD_RegionEco_RegGdpTL2, pre_formatted = TRUE)
PiCork_Eco_OECD_RegionEco_RegGdpTL2 <-get_dataset("REGION_ECONOM", filter=filter_PiCork_Eco_OECD_RegionEco_RegGdpTL2, pre_formatted = TRUE)
PiAntal_Eco_OECD_RegionEco_RegGdpTL2 <-get_dataset("REGION_ECONOM", filter=filter_PiAntal_Eco_OECD_RegionEco_RegGdpTL2, pre_formatted = TRUE)
PiThess_Eco_OECD_RegionEco_GDPLT3 <-get_dataset("REGION_ECONOM", filter=filter_PiThess_Eco_OECD_RegionEco_GDPLT3, pre_formatted = TRUE)
PiAntw_Eco_OECD_RegionEco_GDPLT3 <-get_dataset("REGION_ECONOM", filter=filter_PiAntw_Eco_OECD_RegionEco_GDPLT3, pre_formatted = TRUE)
PiCork_Eco_OECD_RegionEco_GDPLT3 <-get_dataset("REGION_ECONOM", filter=filter_PiCork_Eco_OECD_RegionEco_GDPLT3, pre_formatted = TRUE)
PiAntal_Eco_OECD_RegionEco_GDPLT3 <-get_dataset("REGION_ECONOM", filter=filter_PiAntal_Eco_OECD_RegionEco_GDPLT3, pre_formatted = TRUE)
PiThess_Eco_OECD_RegionEco_RegEmIndu <-get_dataset("REGION_ECONOM", filter=filter_PiThess_Eco_OECD_RegionEco_RegEmIndu, pre_formatted = TRUE)
PiAntw_Eco_OECD_RegionEco_RegEmIndu <-get_dataset("REGION_ECONOM", filter=filter_PiAntw_Eco_OECD_RegionEco_RegEmIndu, pre_formatted = TRUE)
PiCork_Eco_OECD_RegionEco_RegEmIndu <-get_dataset("REGION_ECONOM", filter=filter_PiCork_Eco_OECD_RegionEco_RegEmIndu, pre_formatted = TRUE)
PiAntal_Eco_OECD_RegionEco_RegEmIndu <-get_dataset("REGION_ECONOM", filter=filter_PiAntal_Eco_OECD_RegionEco_RegEmIndu, pre_formatted = TRUE)
PiThess_Eco_OECD_RegionEco_RegGVAWorker <-get_dataset("REGION_ECONOM", filter=filter_PiThess_Eco_OECD_RegionEco_RegGVAWorker, pre_formatted = TRUE)
PiAntw_Eco_OECD_RegionEco_RegGVAWorker <-get_dataset("REGION_ECONOM", filter=filter_PiAntw_Eco_OECD_RegionEco_RegGVAWorker, pre_formatted = TRUE)
PiCork_Eco_OECD_RegionEco_RegGVAWorker <-get_dataset("REGION_ECONOM", filter=filter_PiCork_Eco_OECD_RegionEco_RegGVAWorker, pre_formatted = TRUE)
PiAntal_Eco_OECD_RegionEco_RegGVAWorker <-get_dataset("REGION_ECONOM", filter=filter_PiAntal_Eco_OECD_RegionEco_RegGVAWorker, pre_formatted = TRUE)
PiThess_Eco_OECD_RegionEco_RegIncPC <-get_dataset("REGION_ECONOM", filter=filter_PiThess_Eco_OECD_RegionEco_RegIncPC, pre_formatted = TRUE)
PiAntw_Eco_OECD_RegionEco_RegIncPC <-get_dataset("REGION_ECONOM", filter=filter_PiAntw_Eco_OECD_RegionEco_RegIncPC, pre_formatted = TRUE)
PiCork_Eco_OECD_RegionEco_RegIncPC <-get_dataset("REGION_ECONOM", filter=filter_PiCork_Eco_OECD_RegionEco_RegIncPC, pre_formatted = TRUE)
PiAntal_Eco_OECD_RegionEco_RegIncPC<-get_dataset("REGION_ECONOM", filter=filter_PiAntal_Eco_OECD_RegionEco_RegIncPC, pre_formatted = TRUE)


Timelines.py
#!/usr/bin/env python
# encoding: utf-8

import tweepy #https://github.com/tweepy/tweepy
import csv
import sys


import xlsxwriter
import tweepy 

#https://github.com/tweepy/tweepy

#Twitter API credentials
CONSUMER_KEY = ""
CONSUMER_SECRET = ""
ACCESS_TOKEN = ""
ACCESS_TOKEN_SECRET = ""


def get_all_tweets(user_id):

    auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
    auth.set_access_token(access_token, access_token_secret)
    api = tweepy.API(auth)

    alltweets = []  
    new_tweets = []
    outtweets = []

    new_tweets = api.user_timeline(user_id = user_id,count=200)

    alltweets.extend(new_tweets)

	#save the id of the oldest tweet less one
    oldest = alltweets[-1].id - 1

    #keep grabbing tweets until there are no tweets left to grab
    while len(new_tweets) > 0:
        print "getting tweets before %s" % (oldest)

        #all subsiquent requests use the max_id param to prevent duplicates
        new_tweets = api.user_timeline(user_id = user_id,count=200,max_id=oldest)

        #save most recent tweets
        alltweets.extend(new_tweets)

        #update the id of the oldest tweet less one
        oldest = alltweets[-1].id - 1

        print "...%s tweets downloaded so far" % (len(alltweets))

    #transform the tweepy tweets into a 2D array
    outtweets = [[tweet.id_str, tweet.created_at, tweet.coordinates,tweet.geo,tweet.source,tweet.text] for tweet in alltweets]

    return outtweets

def write_worksheet(twitter_name):

	#formating for excel
	format01 = workbook.add_format()
	format02 = workbook.add_format()
	format03 = workbook.add_format()
	format04 = workbook.add_format()
	format01.set_align('center')
	format01.set_align('vcenter')
	format02.set_align('center')
	format02.set_align('vcenter')
	format03.set_align('center')
	format03.set_align('vcenter')
	format03.set_bold()
	format04.set_align('vcenter')
	format04.set_text_wrap()

	out1 = []
	header = ["id","created_at","coordinates-x","coordinates-y","source","text"]

	worksheet = workbook.add_worksheet(twitter_name)

	out1 = get_all_tweets(twitter_name)
	row = 0
	col = 0

	worksheet.set_column('A:A', 20)
	worksheet.set_column('B:B', 18)
	worksheet.set_column('C:C', 13)
	worksheet.set_column('D:D', 13)
	worksheet.set_column('E:E', 20)
	worksheet.set_column('F:F', 120)

	for h_item in header:
		worksheet.write(row, col, h_item, format03)
		col = col + 1

	row += 1
	col = 0
	
	for o_item in out1:
		write = []
		cord1 = 0
		cord2 = 0
		write = [o_item[0], o_item[1], o_item[4], o_item[5]]

		if o_item[2]:
			cord1 = o_item[2]['coordinates'][0]
			cord2 = o_item[2]['coordinates'][1]
		else:
			cord1 = ""
			cord2 = ""

		format01.set_num_format('yyyy/mm/dd hh:mm:ss')
		worksheet.write(row, 0, write[0], format02)
		worksheet.write(row, 1, write[1], format01)
		worksheet.write(row, 2, cord1, format02)
		worksheet.write(row, 3, cord2, format02)
		worksheet.write(row, 4, write[2], format02)
		worksheet.write(row, 5, write[3], format04)
		row += 1
		col = 0

workbook = xlsxwriter.Workbook('Twitter_timeline.xlsx')


write_worksheet('86307737')
write_worksheet('19122188')
write_worksheet('105285570')
write_worksheet('14310469')
write_worksheet('40566493')
write_worksheet('233245825')
write_worksheet('213793118')


workbook.close()

Twitter/Twitter_keywords.py
from __future__ import print_function

import smtplib

import time
import tweepy
import json
from pymongo import MongoClient

MONGO_HOST = 'mongodb://localhost/twitterdb_guns'  # assuming you have mongoDB installed locally
# and a database called 'seniors'

WORDS = ['bomb explosion', 'bomb explodes', 'gun fire', 'gunshot fire','gunman','stabbing','knife stabs','knife stabbing'] #add keywords to filter tweets
LANGUAGES=['en']

CONSUMER_KEY = ""
CONSUMER_SECRET = ""
ACCESS_TOKEN = ""
ACCESS_TOKEN_SECRET = ""

gmail_user = "youreamil@gmail.com"  #Your email address
gmail_pwd = "password" # Your  email password
FROM = "yourmail@gmail.com" #Your email address
TO = ['yourmail@gmail.com', 'moremail@gmail.com']  # must be a list




def sendMail(SUBJECT,TEXT):


    # Prepare actual message
    message = """\From: %s\nTo: %s\nSubject: %s\n\n%s
    """ % (FROM, ", ".join(TO), SUBJECT, TEXT)
    try:

        server = smtplib.SMTP("smtp.gmail.com", 587)  # or port 465 doesn't seem to work!
        server.ehlo()
        server.starttls()
        server.login(gmail_user, gmail_pwd)
        server.sendmail(FROM, TO, message)
        # server.quit()
        server.close()
        print('successfully sent the mail')
    except:
        print("failed to send mail")



class StreamListener(tweepy.StreamListener):
    # This is a class provided by tweepy to access the Twitter Streaming API.

    def on_connect(self):
        # Called initially to connect to the Streaming API
        print("You are now connected to the streaming API.")

    def on_error(self, status_code):
        # On error - if an error occurs, display the error / status code
        print('An Error has occured: ' + repr(status_code))
        sendMail("WHAT AGAIN?", 'An Error has occured: ' + repr(status_code)) #  Sends mail in case of error
        return False

    def on_data(self, data):
        # This is the meat of the script...it connects to your mongoDB and stores the tweet
        try:
            client = MongoClient(MONGO_HOST)

            # Use seniors database. If it doesn't exist, it will be created.
            db = client.twitterdb_guns

            # Decode the JSON from Twitter
            datajson = json.loads(data)

            # grab the 'created_at' data from the Tweet to use for display
            created_at = datajson['created_at']

            # print out a message to the screen that we have collected a tweet
            print("Tweet collected at " + str(created_at))

            # insert the data into the mongoDB into a collection called twitter_search
            # if twitter_search doesn't exist, it will be created.
            db.twitter_search.insert(datajson)
        except Exception as e:
            print(e)
            # sendMail('DATABASE PROBLEM',e ) # Sends email in case of database failure


while True:
    try:
        auth = tweepy.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET)
        auth.set_access_token(ACCESS_TOKEN, ACCESS_TOKEN_SECRET)
        # Set up the listener. The 'wait_on_rate_limit=True' is needed to help with Twitter API rate limiting.
        listener = StreamListener(api=tweepy.API(wait_on_rate_limit=True))
        streamer = tweepy.Stream(auth=auth, listener=listener)
        print("Tracking: " + str(WORDS))
        streamer.filter(track=WORDS, languages=LANGUAGES)

    # Stops the program when you type Ctrl+C
    except KeyboardInterrupt:
        print ('GOODBYE!!!!!!!!!!!!')
        break
    # Restarts the programme in case of exception  waiting 5sec for every restart
    except:
        time.sleep(10)
        continue



Twitter/GeoBox.py
from __future__ import print_function

import smtplib

import time
import tweepy
import json
from pymongo import MongoClient

MONGO_HOST = 'mongodb://localhost'  # assuming you have mongoDB installed locally



CONSUMER_KEY = ""
CONSUMER_SECRET = ""
ACCESS_TOKEN = ""
ACCESS_TOKEN_SECRET = ""

gmail_user = "youreamil@gmail.com"  #Your email address
gmail_pwd = "password" # Your  email password
FROM = "yourmail@gmail.com" #Your email address
TO = ['yourmail@gmail.com', 'moremail@gmail.com']  # must be a list



def sendMail(SUBJECT, TEXT):
    # Prepare actual message
    message = """\From: %s\nTo: %s\nSubject: %s\n\n%s
    """ % (FROM, ", ".join(TO), SUBJECT, TEXT)
    try:

        server = smtplib.SMTP("smtp.gmail.com", 587)  # or port 465 doesn't seem to work!
        server.ehlo()
        server.starttls()
        server.login(gmail_user, gmail_pwd)
        server.sendmail(FROM, TO, message)
        # server.quit()
        server.close()
        print('successfully sent the mail')
    except:
        print("failed to send mail")


class StreamListener(tweepy.StreamListener):
    # This is a class provided by tweepy to access the Twitter Streaming API.

    def on_connect(self):
        # Called initially to connect to the Streaming API
        print("You are now connected to the streaming API.")

    def on_error(self, status_code):
        # On error - if an error occurs, display the error / status code
        print('An Error has occured: ' + repr(status_code))
        sendMail("WHAT AGAIN?", "OTHER EXCEPTION")  # Sends mail in case of error
        return False

    def on_data(self, data):
        # This is the meat of the script...it connects to your mongoDB and stores the tweet
        try:
            client = MongoClient(MONGO_HOST)

            # Use CHICAGO database. If it doesn't exist, it will be created.
            db = client.CHICAGO

            # Decode the JSON from Twitter
            datajson = json.loads(data)

            # grab the 'created_at' data from the Tweet to use for display
            created_at = datajson['created_at']

            # print out a message to the screen that we have collected a tweet
            print("Tweet collected at " + str(created_at))

            # insert the data into the mongoDB into a collection called twitter_search
            # if twitter_search doesn't exist, it will be created.
            db.CHICAGO.insert(datajson)
        except Exception as e:
            print(e)
            sendMail('DATABASE PROBLEM', " Database Exception")  # Sends email in case of database failure


while True:
    try:
        auth = tweepy.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET)
        auth.set_access_token(ACCESS_TOKEN, ACCESS_TOKEN_SECRET)
        # Set up the listener. The 'wait_on_rate_limit=True' is needed to help with Twitter API rate limiting.
        listener = StreamListener(api=tweepy.API(wait_on_rate_limit=True))
        streamer = tweepy.Stream(auth=auth, listener=listener)
        # print("Tracking: " + str(WORDS))
        print("Tracking tweets from Chicago")

        GEOBOX_CHICAGO = [-87.9671519386,41.62241735,-87.3972361672,42.036640212] # here you add your geobox
        streamer.filter(locations=GEOBOX_CHICAGO)
    # Stops the program when you type Ctrl+C
    except KeyboardInterrupt:
        print('GOODBYE!!!!!!!!!!!!')
        break
    # Restarts the programme in case of exception  waiting 5sec for every restart
    except Exception as e:
        print(e)
        time.sleep(10)
        continue

